src/Compilers/Core/CodeAnalysisTest/PDB/CustomDebugInfoReaderTests.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Debugging;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests;
+
+public class CustomDebugInfoReaderTests
+{
+    [Theory]
+    [InlineData(new byte[0], "")]
+    [InlineData(new byte[] { 0x00, 0x00 }, "")]
+    [InlineData(new byte[] { (byte)'a', 0x00 }, "a")]
+    [InlineData(new byte[] { (byte)'a', 0x00, 0x00, 0x00 }, "a")]
+    public void DecodeForwardIteratorRecord(byte[] bytes, string expected)
+    {
+        Assert.Equal(expected, CustomDebugInfoReader.DecodeForwardIteratorRecord(bytes.ToImmutableArray()));
+    }
+
+    [Theory]
+    [InlineData(new byte[] { 0x00 })]
+    [InlineData(new byte[] { (byte)'a', })]
+    [InlineData(new byte[] { (byte)'a', 0x00, 0x00 })]
```

Comment:
>{ (byte)'a', 0x00, 0x00 } [](http://example.com/codeflow?start=27&length=25)

Why is this case and the `{ 0x00 }` case invalid, but not cases with explicit `0x00` in the previous test?

Reply:
The blob needs to be valid UTF16 string.